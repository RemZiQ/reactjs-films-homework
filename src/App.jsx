import React from 'react';
import { Route, Switch } from 'react-router-dom';
import propTypes from 'prop-types';

import MovieDetailsPage from './components/MovieDetailsPage';
import MovieList from './components/MoviewList';


const App = ({ location }) => {
  console.log(location);
  const params = new URLSearchParams(location.search);
  return (
    <div>
      <MovieDetailsPage query={params.get('query')} />
      <Switch>
        <Route path="/" exact component={MovieList} />
        <Route path="/:search" component={MovieList} />
      </Switch>
    </div>
  );
};

App.propTypes = {
  location: propTypes.objectOf(propTypes.any).isRequired,
};

export default App;
