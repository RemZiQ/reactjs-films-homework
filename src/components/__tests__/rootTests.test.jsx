import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import App from '../../App';

it('App renders well', () => {
  const props = {
    location: { search: '' },
  };
  const renderer = new ShallowRenderer();
  const result = renderer.render(<App {...props} />);
  expect(result).toMatchSnapshot();
});
