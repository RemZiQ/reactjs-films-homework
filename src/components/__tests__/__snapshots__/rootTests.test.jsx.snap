// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders well 1`] = `
<div>
  <Connect(MovieDetailsPage)
    query={null}
  />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/:search"
    />
  </Switch>
</div>
`;
